README.md

```diff
@@ -1,77 +1,119 @@
+
+```ts
+private syncSdkReady(context: InitContext) {
+  // implement your application start-up logic        
+}
+```
+
+## InitContext
+InitContext contains data that can be used by your application to target specific interview or user.
+
+```ts
+// Unique addin session shared by all the users.
+addinSessionId: string;
+
+// Unique has of a user identifier he has in a given addin session.
+addinSessionUserId: string;
+
+// Interview session under which addins are executing.
+sessionId: string;
+
+// Authorization token which is sent to the addin API.
+token: string;
+
+// User's language used in the session.
+language: string;
+
+// An array of addin configuration values (optional).
+configuration?: ConfigurationValue[];
 ```
 
 ## Broadcast actions
 
-When our notepad wants to send out new actions to the other instances, the app just calls `send(type, payload)` to broadcast its information. 
+When our notepad wants to send out new actions to the other instances, the app needs to explicitly connect to the signaling service. 
+
+```ts
+Sync.connect().then(() => {
+  // you are now connected and you can send messages
+});
+```
+
+After that your application can just call `Sync.sendMessage(type: string, payload?: any);` to broadcast its information. 
+- `type` can be any string, identifying the type of your message in the application, so in our case of note pad it can be 'ADD_NOTE'.
+- `payload` is optional parameter and can be any payload that is sent to the other participants that are sharing same `addinSessionId` (~all participants using same addin in same interview session).
 
 ```js
-Sync.send('ADD_NOTE', {
-  message: textInput.value,
-  author: this.state.author
-})
+Sync.sendMessage('ADD_NOTE', {
+  note: textInput.value
+});
 ```
 
+### NOTE
+There are certain limits on how many messages can be sent in time and what can be overal size of the messages:
+- only 50 messages can be sent every 200ms
+- total amount of data that can be sent every 200ms cannot be higher than 128Kb
+
+In case that the limit is reached and next message is sent we are informing aplication using the `errorHandler` (please see below);
+
 ## Receive actions
 
-To handle incoming messages the app needs to pass an event handler to the `onReceive((type, payload) => any)` function.
+To handle incoming messages the app needs to register the message handler.
 
-When our notepad receives a new action, we can do a switch case on the action type. Then we divert it to the correct function to handle the payload of the action.
+```ts
+import Sync, { Message } from 'skype-sync';
 
-```js
-Sync.onReceive((type, payload) => {
-  switch (type) {
-    case 'ADD_NOTE':
-      // handle adding new note
-      addNote(payload)
-      break
-    case 'DELETE_NOTE':
-      // handle deleting a note
-      deleteNote(payload)
-      break
-    default:
-      return
-  }
-})
+Sync.messageHandler = (messageRequest: Message) => {
+  // handle incoming message
```

Comment:
I'd show a switch-case statement here based on message type ADD_NOTE from previous section to make it even more obvious